src/modules/flow/power-supply/power-supply.json

```diff
@@ -0,0 +1,166 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "power-supply",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "BSD-3-Clause",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "power-supply",
+      "description": "Send list of all power supplies attached to board.",
+      "in_ports": [
+        {
+          "data_type": "any",
+          "description": "Trigger to get list of power supplies.",
+          "methods": {
+            "process": "get_list_process"
+          },
+          "name": "TRIGGER"
+        },
+        {
+          "data_type": "string",
+          "description": "Name of type to be used to filter power supplies. It can be 'Any', 'Unknown', 'Battery', 'UPS', 'Mains', 'USB', 'USB_DCP', 'USB_CDP', 'USB_ACA'.",
+          "methods": {
+            "process": "set_type_process"
+          },
+          "name": "TYPE"
+        }
+      ],
+      "methods": {
+        "open": "get_list_open"
+      },
+      "name": "power-supply/get-list",
```

Comment:
I completely agree on keeping consistency between node type names, but I wasn't able to find specific nodes to get lists or scan stuff... oic, for example, doesn't have a scan / discovery node. It's one of it clients port.

I can change the name from "get-list" to "discover" if it seems more descriptive, but apparently it'll be the first of this kind anyway.